flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceWithWatermarkExtractor.java

```diff
@@ -79,6 +80,12 @@ public class TestIcebergSourceWithWatermarkExtractor implements Serializable {
   @TempDir protected Path temporaryFolder;
 
   private static final InMemoryReporter REPORTER = InMemoryReporter.createWithRetainedMetrics();
+  public static final Configuration DISABLE_CLASSLOADER_CHECK_CONFIG =
+      new Configuration()
+          // disable classloader check as Avro may cache class/object in the serializers.
+          .set(CoreOptions.CHECK_LEAKED_CLASSLOADER, false)
+          // disable inferring source parallelism
```

Comment:
inferred parallelism might mess up the watermark and record ordering comparison. disable it to avoid the flakiness

Reply:
Why is this so?
Do you have any idea? Would it be an issue in prod?

Reply:
good question. let me dig more.

Reply:
`TestIcebergSourceSql` assume the parallelism is 1 for `testWatermarkOptionsAscending` and `testWatermarkOptionsDescending`. Table has 2 files. This test just check split assignment is ordered with single reader.

```
    tableEnvironment.getConfig().set("table.exec.resource.default-parallelism", "1");
```

Reply:
Maybe explicitly setting the parallelism in those tests would be better.
WDYT?

Reply:
actually, we don't need the disabling for this particular test as it doesn't go through the `buildStream(env)` path where infer parallelism happens. will revert the change